src/qibo/tests/regressions/trotter_adiabatic_bfgs.out

```diff
@@ -1,2 +1,2 @@
-5.145435393136968960e-01
-2.009895284405383897e+00
+5.149684200950115365e-01
```

Comment:
Any special motivation to update this file?

Reply:
This PR changes how the `TrotterHamiltonian` is constructed from symbols and the `make_compatible` so the grouping of terms in the Trotter decomposition may be slightly different and thus lead to slightly different evolution (with the difference being smaller than the expected Trotter error). This also changes slightly the optimal parameters found by this optimization test. The difference between old and new values is small (<1e-3) so I believe it can be explained from this.